tests/compile.sh

```diff
@@ -3,42 +3,171 @@ set -eux
+readonly BUILD_TARGET=$2
+readonly MAKE_OPT=${3:-}
+readonly BUILD_NAME=fv3_mom6_cice${4:+_$4}
+
+readonly clean_before=${5:-YES}
+readonly clean_after=${6:-YES}
 
 hostname
 
-readonly compile_root_dir=$(cd "$(dirname "$0")" && pwd -P )
-PATHTR=${PATHTR:-$(cd ${compile_root_dir}/.. && pwd)}
-echo "PATHTR is $PATHTR"
-cd ${PATHTR}/NEMS
+# ----------------------------------------------------------------------
+
+echo "Compiling ${MAKE_OPT} into $BUILD_NAME.exe on $BUILD_TARGET"
+
+# ----------------------------------------------------------------------
+# Make sure we have a "make" and reasonable threads.
+
+gnu_make=gmake
+if ( ! which $gnu_make ) ; then
+    echo WARNING: Cannot find gmake in \$PATH.  I will use \"make\" instead.
+    gnu_make=make
+    if ( ! $gnu_make --version 2>&1 | grep -i gnu > /dev/null 2>&1 ) ; then
+       echo WARNING: The build system requires GNU Make. Things may break.
+    fi
+fi
+
+if [[ $BUILD_TARGET == cheyenne.* || $BUILD_TARGET == stampede.* ]] ; then
+    MAKE_THREADS=${MAKE_THREADS:-3}
+fi
+
+# CMEPS component fails to build when using multiple threads, thus set to 1
+MAKE_THREADS=${MAKE_THREADS:-1}
+
+if [[ "$MAKE_THREADS" -gt 1 ]] ; then
+    echo Using \$MAKE_THREADS=$MAKE_THREADS threads to build FV3 and FMS.
+    echo Consider reducing \$MAKE_THREADS if you hit memory or process limits.
+    gnu_make="$gnu_make -j $MAKE_THREADS"
+    #gnu_make="$gnu_make"
+fi
+
+# ----------------------------------------------------------------------
+# Configure NEMS and components
+
+# Configure NEMS
+cd "$PATHTR/../NEMS"
+
+COMPONENTS="FMS,FV3"
+if [[ "${MAKE_OPT}" == *"CCPP=Y"* ]]; then
+  COMPONENTS="$COMPONENTS,CCPP"
+
+  # Check if suites argument is provided or not
+  set +ex
+  TEST=$( echo $MAKE_OPT | grep -e "SUITES=" )
+  if [[ $? -eq 1 ]]; then
+    echo "No suites argument provided, compiling all available suites ..."
+    # Loop through all available suite definition files and extract suite names
+    SDFS=(../FV3/ccpp/suites/*.xml)
+    SUITES=""
+    for sdf in ${SDFS[@]}; do
+      suite=${sdf#"../FV3/ccpp/suites/suite_"}
+      suite=${suite%".xml"}
+      SUITES="${SUITES},${suite}"
+    done
+    # Remove leading comma
+    SUITES=${SUITES#","}
+  else
+    SUITES=$( echo $MAKE_OPT | sed 's/.* SUITES=//' | sed 's/ .*//' )
+  fi
+  echo "Compiling suites ${SUITES}"
+  MAKE_OPT="${MAKE_OPT} SUITES=${SUITES}"
+  set -ex
+
+  # FIXME - create CCPP include directory before building FMS to avoid
+  # gfortran warnings of non-existent include directory (adding
+  # -Wno-missing-include-dirs) to the GNU compiler flags does not work,
+  # see also https://gcc.gnu.org/bugzilla/show_bug.cgi?id=55534);
+  # this line can be removed once FMS becomes a pre-installed library
+  mkdir -p $PATHTR/ccpp/include
+fi
 
-if [[ ${app_name} =~ _debug ]]; then
-  S2S_DEBUG_MODULE=true
+if [[ "${MAKE_OPT}" == *"MOM6=Y"* ]]; then
+  COMPONENTS="$COMPONENTS,MOM6"
+fi
+if [[ "${MAKE_OPT}" == *"CICE=Y"* ]]; then
```

Comment:
Since we will shortly be bringing in CICE6, it might be helpful to just insert the option for CICE6 here?